### ISSUE TITLE
[java] CommentDefaultAccessModifier: false positive with TestNG annotations

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4268]
**Affects PMD Version:6.50.0**
 
**Rule:** [CommentDefaultAccessModifier](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#commentdefaultaccessmodifier)

**Description:**
In the rule CommentDefaultAccessModifier, I found PMD can raise a wanring for `org.testng.annotations.Test`, but when I used `org.junit.jupiter.api.Test`, it does not report the warning, hence I think this is a false positive because these two annotations are equal and the behavior should be consistent.

**Code Sample demonstrating the issue:**

```java
import org.testng.annotations.Test;
public class C {
    @Test
    void foo() {
        synchronized (this) {}
    }
}
```

**Expected outcome:**

PMD should not report a violation at line 3. This is a false positive.

**Running PMD through:** *[Maven]*

**Comment by adangel** [comment]
> these two annotations are equal and the behavior should be consistent

Only if the two test frameworks behave the same. In the official [TestNG documentation](https://testng.org/doc/documentation-main.html), all examples use public methods. But it seems, that default access also works with TestNG.

Please also note, that if `@Test` is used as a class annotation, only _public_ methods will be considered: https://testng.org/doc/documentation-main.html#class-level

> The effect of a class level `@Test` annotation is to make all the public methods of this class to become test methods even if they are not annotated.
